### PR TITLE
Squash all intake model migrations

### PR DESCRIPTION
**Pull request by jennchenn**
## Notion ticket link

* [Squash migrations from sprint 1](https://www.notion.so/uwblueprintexecs/Squash-migrations-from-Sprint-1-4b86a1c04f6f44c58c1ec53100e3d8c9)
* [Add reference to child in caregivers model](https://www.notion.so/uwblueprintexecs/Add-reference-to-child-in-caregivers-model-71ec1d78cf4b4d5b9638d6ec7fd5b8a8)
* [Create children model](https://www.notion.so/uwblueprintexecs/Create-children-model-031ad31e425d4b0f9151aee5182c9b42)
* [Create visit locations model](https://www.notion.so/uwblueprintexecs/Create-visit_locations-model-89fec44fb1404e4e8b9c8f63fde32f2d)
* [Create intakes model](https://www.notion.so/uwblueprintexecs/Create-intakes-model-a28001141c9c427d96a17a87faa6e6d1)
* [Create intake<>concerns association tables](https://www.notion.so/uwblueprintexecs/Create-association-tables-for-intake-child-concerns-and-intake-familial-concerns-b5d08078495240fb90491d17de0ab12d)
* [Create short/long term goals models](https://www.notion.so/uwblueprintexecs/Create-short-long-term-goals-models-4c8e0ae6e0454d5db22efe4f414ca4a6)

## Implementation description
* Added in the remaining tables required for creating intakes
* Squashed all migrations into one for easier management

## Steps to test
1. Restart your docker containers: `docker-compose down --volumes && docker-compose up`
2. Exec into the backend container: `docker exec -it cas_py_backend /bin/bash`
3. Upgrade the database: `flask db upgrade` -- verify that it works successfully
4. Downgrade and upgrade the database again: `flask db downgrade && flask db upgrade` -- verify that it works successfully
5. Exec into the db container: `docker exec -it cas_db psql -U postgres -d cas`
6. Verify that you see the following tables (there should be 16 + the `alembic_version` table):
```
                   List of relations
 Schema |           Name            | Type  |  Owner
--------+---------------------------+-------+----------
 public | access_types              | table | postgres
 public | addresses                 | table | postgres
 public | alembic_version           | table | postgres
 public | branches                  | table | postgres
 public | caregivers                | table | postgres
 public | child_concerns            | table | postgres
 public | children                  | table | postgres
 public | daytime_contacts          | table | postgres
 public | familial_concerns         | table | postgres
 public | intakes                   | table | postgres
 public | intakes_child_concerns    | table | postgres
 public | intakes_familial_concerns | table | postgres
 public | long_term_goals           | table | postgres
 public | short_term_goals          | table | postgres
 public | transportation_methods    | table | postgres
 public | users                     | table | postgres
 public | visit_locations           | table | postgres
```
7. Check that the tests still run successfully: `docker exec -it cas_py_backend /bin/bash -c "pip install -e . && pytest"`

## What should reviewers focus on?
* does the database still look okay?

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

**Review by andrew-y-ma — COMMENTED** [review]
Left some comments/questions. Looks great!

File: backend/python/app/models/base_mixin.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+from sqlalchemy import inspect
+from sqlalchemy.orm.properties import ColumnProperty
+
+
+class BaseMixin(object):
+    def to_dict(self, include_relationships=False):
+        # define the address table
```
[the remark this answers is not in the document]
Comment (jennchenn) [reply backend/python/app/models/base_mixin.py]: oops nice catch

File: backend/python/app/models/child.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    cpin_number = db.Column(db.String, nullable=True)
+    child_service_worker_id = db.Column(
+        db.Integer, db.ForeignKey("users.id"), nullable=False
+    )
+    daytime_contact_id = db.Column(
+        db.Integer, db.ForeignKey("daytime_contacts.id"), nullable=False
+    )
+    special_needs = db.Column(db.String, nullable=False)
+    has_kinship_provider = db.Column(db.Boolean, nullable=False)
+    has_foster_placement = db.Column(db.Boolean, nullable=False)
```
[the remark this answers is not in the document]
Comment (jennchenn) [reply backend/python/app/models/child.py:23]: added comment!

File: backend/python/app/models/visit_location.py [hunk 3]
```diff
@@ -0,0 +1,9 @@
+from . import db
+from .base_mixin import BaseMixin
+
+
+class VisitLocation(db.Model, BaseMixin):
+    __tablename__ = "visit_locations"
+
+    id = db.Column(db.Integer, primary_key=True, nullable=False)
+    location = db.Column(db.String, nullable=False)
```
[the remark this answers is not in the document]
Comment (jennchenn) [reply backend/python/app/models/visit_location.py:9]: bad documentation on my part - will update arch design doc but this should be fine for now

[the remark this answers is not in the document]
Comment (andrew-y-ma) [reply backend/python/app/models/visit_location.py:9]: Cool sg!

**Review by andrew-y-ma — APPROVED** [review]
Nice work!!

File: backend/python/app/models/child.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+class Child(db.Model, BaseMixin):
+    __tablename__ = "children"
+
+    id = db.Column(db.Integer, primary_key=True, nullable=False)
+    intake_id = db.Column(db.Integer, db.ForeignKey("intakes.id"), nullable=True)
+    first_name = db.Column(db.String, nullable=False)
+    last_name = db.Column(db.String, nullable=False)
+    date_of_birth = db.Column(db.Date, nullable=True)
+    cpin_number = db.Column(db.String, nullable=True)
```
Comment (Het7) [inline backend/python/app/models/child.py:13]: Question: Does this have to be a string or can it be an integer?

Comment (Het7) [inline backend/python/app/models/child.py:13]: Question: Does this have to be a string? Or does it make more sense to keep it as a number?

**Comment by Het7** [comment]
Great work!

**Review by Het7 — APPROVED** [review]

**Comment by jennchenn** [comment]
thanks for the reviews!!

File: backend/python/app/models/child.py (hunk 4 above)
Comment (jennchenn) [reply backend/python/app/models/child.py:13]: oops sorry forgot to respond - the cpin number is sort of like an id so I think we can just keep as a string for now

**Pull request merged**